labelbox/schema/model_run.py

```diff
@@ -80,27 +80,40 @@ def upsert_labels(self, label_ids, timeout_seconds=3600):
             }})['MEALabelRegistrationTaskStatus'],
                                      timeout_seconds=timeout_seconds)
 
-    def upsert_data_rows(self, data_row_ids, timeout_seconds=3600):
+    def upsert_data_rows(self,
+                         data_row_ids=None,
+                         global_keys=None,
+                         timeout_seconds=3600):
         """ Adds data rows to a Model Run without any associated labels
         Args:
-            data_row_ids (list): data row ids to add to mea
+            data_row_ids (list): data row ids to add to model run
+            global_keys (list): global keys for data rows to add to model run
             timeout_seconds (float): Max waiting time, in seconds.
         Returns:
             ID of newly generated async task
         """
+        if data_row_ids is None == global_keys is None:
```

Comment:
Do we need this validation if we're handling it on the backend?

Reply:
Might not be a bad idea to validate this on the resolver. Would encourage us to remember to include the validation in the controller in SDK4 as well. In SDK4 we should probably include this into OpenAPI based validation, and implement something like `openapi-typescript-validator` (or not sure if there's anything more native) on the controllers so this is all defined declaratively and we can generate documentation from it.

Reply:
This is already being validated in the resolver so I have removed it from here.